dna/pDna/pDna.go

```diff
@@ -0,0 +1,78 @@
+package pDna
+
+import (
+	"log"
+	"math"
+)
+
+// Float32Base encodes a DNA base as a probability vector using float32 precision.
+// gap probabilities can be stored implicitly as 1 - (A + C + G + T).
+type Float32Base struct {
+	A float32
+	C float32
+	G float32
+	T float32
+}
+
+// Uint8Base encodes a DNA base as a probability vector using uint8 probability encoding.
+// A uint8 value x can be converted to a probability as follows: p = 1 / (2^(x/10)).
+// gap probabilities can be stored implicitly as 1 - (A + C + G + T).
+type Uint8Base struct {
+	A uint8
+	C uint8
+	G uint8
+	T uint8
+}
+
+type Uint32Base uint32
+
+// Pack takes an input base in Uint8Base format (where each base probability A, C, G, and T is stored in a uint8)
+// and converts it to a single uint32Base, where each consecutive 8 bits corresponds to the probability of one of the
+// four bases.
+func Pack(base Uint8Base) Uint32Base {
```

Comment:
for the uint8base idea I don't think the un/pack functions are necessary. The struct is 32 bit anyways so why not just leave it in struct form? 

Though if we go with Craig's idea of a trimmed down float then conversion functions will be necessary.

Reply:
A few considerations:
1. Internally, we might always convert to floats for most use cases, so the uint representations may just be for reading and writing to disk.
2. If that's the case, we'll probably never use the Uint8Base struct and want to convert directly between the Uint32Base and Float32Base. 
3. So with that said, I'm supportive of removing either the Uint32Base or the Uint8Base struct/type, to eliminate redundancy. I think either solution makes sense to me, though I assume the Float32Base is slightly more lightweight

Reply:
I agree, sounds like a great plan! I/O from a binary file format into that struct should be straightforward as well